### ISSUE TITLE
github-pages deploy error

### ISSUE THREAD

**Issue by JacquesCarette** [JacquesCarette/Drasil#3483]
The latest deploy didn't work - @balacij could you investigate?

**Comment by balacij** [comment]
Sorry, where do you see the error? I do see a _huge_ issue with compiling the TeX artifacts -- it seems that the time to compile them takes up nearly 18min!

Latest run I see on `master`: https://github.com/JacquesCarette/Drasil/actions/runs/5248415964

**Comment by smiths** [comment]
I don't know if this is the right place to say it, but we probably don't have to compile our tex artifacts every time, especially if the changes don't modify the tex files.  (Maybe the github actions is already smart enough to not rebuild the pdf files when the tex files haven't changed.)  Even if the tex files have been updated, we don't necessarily have to build the pdf files every time.  Most of us don't currently consult the pdf files very often, and if need them, we can build them locally.

I'd rather build everything every time, but 18 minutes seems like it might be problematic.

**Comment by balacij** [comment]
No, you're completely right. We should make it only run if the stable *.tex artifacts are changed. That's a great idea.

**Comment by JacquesCarette** [comment]
To find this: go to the home page, look at the right-hand bar to where it lists "Environments" and it lists github-pages with <error> beside it.  Clicking on [that link](https://github.com/JacquesCarette/Drasil/deployments/activity_log?environment=github-pages) gets to a log page. I have, just now, manually restarted the 'failed' part, but that didn't work - seems that the artefacts it was looking for were not found.

But yeah, 18 minutes is quite excessive.  I think [precompilation](https://tex.stackexchange.com/questions/79493/ultrafast-pdflatex-with-precompiling) might be something worth looking in to. Lots of latex time is spent on packages, so if we can amortize that cost, I think that will bring this down a lot.

**Comment by smiths** [comment]
Is it working now?  I just looked at the Deployments page and it says github-pages is active as of an hour ago.  There were two previous failures.  If this just a GitHub problem (that has not been fixed) and not something we need to investigate?

**Comment by JacquesCarette** [comment]
It's working now, so let's assume it was a glitch.
